src/Features/CSharp/Portable/Completion/CompletionProviders/OperatorIndexerCompletionProvider.cs

```diff
@@ -0,0 +1,234 @@
+        // \uffdc http://www.fileformat.info/info/unicode/char/ffdc/index.htm (non-surrogate)
+        private const string SortingPrefix = "\uFFFD";
+
+        internal override ImmutableHashSet<char> TriggerCharacters => ImmutableHashSet.Create('.');
+
+        internal override bool IsInsertionTrigger(SourceText text, int insertedCharacterPosition, OptionSet options)
+        {
+            var ch = text[insertedCharacterPosition];
+            return ch == '.';
+        }
+
+        protected override Task<CompletionDescription> GetDescriptionWorkerAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+            => SymbolCompletionItem.GetDescriptionAsync(item, document, cancellationToken);
+
+        private static ImmutableDictionary<string, string> CreateCompletionHandlerProperty(string operation, params (string key, string value)[] additionalProperties)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<string, string>();
+            builder.Add(CompletionHandlerPropertyName, operation);
+            foreach (var (key, value) in additionalProperties)
+            {
+                builder.Add(key, value);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var cancellationToken = context.CancellationToken;
+            var document = context.Document;
+            var position = context.Position;
+            var syntaxTree = await document.GetRequiredSyntaxTreeAsync(cancellationToken).ConfigureAwait(false);
+            var token = syntaxTree.FindTokenOnLeftOfPosition(position, cancellationToken);
+            token = token.GetPreviousTokenIfTouchingWord(position);
+            if (!token.IsKind(SyntaxKind.DotToken))
+            {
+                return;
+            }
+
+            var expression = GetParentExpressionOfInvocation(token);
+            if (expression is null)
+            {
+                return;
+            }
+
+            var semanticModel = await document.ReuseExistingSpeculativeModelAsync(expression.SpanStart, cancellationToken).ConfigureAwait(false);
+            var container = semanticModel.GetTypeInfo(expression, cancellationToken).Type;
+            if (container is null)
+            {
+                return;
+            }
+
+            var allMembers = container.GetMembers();
+            var allExplicitConversions = from m in allMembers.OfType<IMethodSymbol>()
+                                         where
+                                             m.IsConversion() && // MethodKind.Conversion
+                                             m.Name == WellKnownMemberNames.ExplicitConversionName && // op_Explicit
+                                             container.Equals(m.Parameters[0].Type) // Convert from container type to other type
```

Comment:
what if hte conversionhas no parameters?  please add test case to make sure we don't crash.

Reply:
You were right. I added a guard and a test in 662c5e35